### ISSUE TITLE
Abstract + using causes methods to apply to the Concrete type.

### ISSUE THREAD

**Issue by m22spencer** [HaxeFoundation/haxe#5888]
When using an abstract, any non-static methods will apply to the concrete type, even if there is no valid `from`/`@:from`.

http://try.haxe.org/#5C204
```haxe
using Test;

class Test {
    static function main() {
        0.bar();   //Should not compile.
    }
}

abstract Foo(Int) {
    public function bar() {};
}
```

Tested on: Haxe Compiler 3.4.0 (git build development @ fd22bf5)

**Comment by Simn** [comment]
Hmm yes, I suppose we should ignore non-static abstract functions here.

**Comment by Simn** [comment]
I'll fix this for 3.4 because it's easy to fix and could cause some problems that are hard to understand for peasants.
